File: DSA/VaniJain_590012020/Day5Question3.java
import java.util.*;
public class Day5Question3 
{
    public static void main(String[] args) 
    {
        System.out.println("Enter the number of elements in the array:");
        Scanner scr = new Scanner(System.in);
        int n = scr.nextInt();
        int[] array = new int[n];
        int i, j;
        System.out.println("Enter the elements into the array:");
        for (i = 0; i < n; i++) 
        {
            array[i] = scr.nextInt();
        }
        boolean flag = false;
        for (i = 0, j = n - 1; i <= j; i++, j--)
        {
            if (array[i] != array[j]) 
            {
                flag = true;
                break;
            }
        }

        if (flag)
            System.out.println("The Array is not a Palindrome");
        else
            System.out.println("The Array is a Palindrome");
    }
}

File: DSA/Nivedan_590014904/Day_5_Question_3.c
#include <stdio.h>
int palindrome(int arr[], int n) {
    int j = n - 1;
    for (int i = 0; i < n / 2; i++) {
        if (arr[i] != arr[j]) {
            return 0;  
        }
        j--;
    }
    return 1;  
}
int main() {
    int n;
    printf("Enter the number of elements: ");
    scanf("%d", &n);
    int arr[n];
    printf("Enter %d elements:\n", n);
    for (int i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }

    if (palindrome(arr, n)) {
        printf("The array is a palindrome.\n");
    } else {
        printf("The array is not a palindrome.\n");
    }
}
